stdlib/src/builtin/string.mojo

```diff
@@ -1406,36 +1408,135 @@ struct String(
             substr._strref_dangerous(), start=start
         )
 
-    fn split(self, delimiter: String) raises -> List[String]:
-        """Split the string by a delimiter.
+    fn split(self, sep: String, maxsplit: Int = -1) -> List[String]:
+        """Split the string by a separator. Defaults to whitespaces and
+        universal newlines [just like Python](
+            https://docs.python.org/3/library/stdtypes.html) EXCEPT
+        for the `u2028` and `u2029` unicode separators.
 
         Args:
-          delimiter: The string to split on.
+            sep: The string to split on.
+            maxsplit: The maximum amount of items to split from String.
+                Defaults to unlimited.
 
         Returns:
-          A List of Strings containing the input split by the delimiter.
+            A List of Strings containing the input split by the separator.
 
-        Raises:
-          Error if an empty delimiter is specified.
-        """
-        if not delimiter:
-            raise Error("empty delimiter not allowed to be passed to split.")
+        Examples:
 
+        ```mojo
+        # Splitting a space
+        _ = String("hello world").split(" ") # ["hello", "world"]
+        # Splitting adjacent separators
+        _ = String("hello,,world").split(",") # ["hello", "", "world"]
+        # Splitting with maxsplit
+        _ = String("1,2,3").split(",", 1) # ['1', '2,3']
+        # Splitting an empty string or filled with whitespaces
+        _ = String("      ").split() # []
+        _ = String("").split() # []
+        # Splitting a string with leading, trailing, and middle whitespaces
+        _ = String("      hello    world     ").split() # ["hello", "world"]
```

Comment:
Please remove these since they describe the other overload (where you already have these).